pic-sure-resources/pic-sure-visualization-resource/src/test/java/edu/harvard/hms/dbmi/avillach/resource/visualization/service/VisualizationUtilTests.java

```diff
@@ -0,0 +1,26 @@
+package edu.harvard.hms.dbmi.avillach.resource.visualization.service;
+
+import edu.harvard.dbmi.avillach.util.VisualizationUtil;
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class VisualizationUtilTests {
+
+    @Test
+    @DisplayName("Test limitKeySize")
+    public void testLimitKeySizeUniqueness() {
+        // Testing name uniqueness
+        Map<String, Integer> axisMap = new HashMap<>();
+        axisMap.put("Disease-Specific (Asthma, Allergy and Inflammation, PUB)", 1);
+        axisMap.put("Disease-Specific (Asthma, Allergy and Inflammation, PUB, NPU)", 1);
+        axisMap.put("Disease-Specific (Asthma, Allergy and Inflammation, NPU)", 1);
+        axisMap.put("Disease-Specific (Asthma, Allergy and Inflammation)", 1);
+
+        Map<String, Integer> stringIntegerMap = VisualizationUtil.limitKeySize(axisMap);
+        assert (stringIntegerMap.size() == 4);
+    }
```

Comment:
I dislike this test for the following reasons:

- You used the raw `assert` rather than using junits. No clean diffs when things break, confusing test reports.
- Your verification step is not thorough. My hunch is that you're verifying the bug that you fixed in this PR, but the unit of logic under test does more than this test lets on. Merely returning the map you pass to `limitKeySize` would cause this test to pass

Here's a test that I think is better:
 ```suggestion
    @Test
    @DisplayName("Test limitKeySize")
    public void testLimitKeySizeUniqueness() {
        Map<String, Integer> axisMap = new HashMap<>(Map.of(
            "Disease-Specific (Asthma, Allergy and Inflammation, PUB)", 1,
            "Disease-Specific (Asthma, Allergy and Inflammation, PUB, NPU)", 1,
            "Disease-Specific (Asthma, Allergy and Inflammation, NPU)", 1,
            "Disease-Specific (Asthma, Allergy and Inflammation)", 1
        ));

        Map<String, Integer> actual = VisualizationUtil.limitKeySize(axisMap);

        Map<String, Integer> expected = Map.of(
            "Disease-Specific (Asthma, Allergy an..., PUB)", 1,
            "Disease-Specific (Asthma, Allergy an...ation)", 1,
            "Disease-Specific (Asthma, Allergy an..., NPU)", 1,
            "Disease-Specific (Asthma, Allergy a...B, NPU)", 1
        );
        assertEquals(expected, actual);
    }
```